src/main/java/seedu/address/ui/AttendanceTag.java

```diff
@@ -16,14 +21,29 @@ private AttendanceTag() {
     /**
      * Produces an attendance tag to be added to {@code PetCard}.
      *
-     * @param attendance the attendance information to be included.
-     * @return a green label if the pet was present on the given date, a red label if the pet was absent.
+     * @param attendanceEntry the attendance to be converted to an attendance tag.
+     * @return a green label if the pet is present on the given date,
+     * a red label if the pet is absent,
+     * and a grey label if no attendance was marked on that day.
      */
-    public static Label createAttendanceTag(String attendance) {
-        String[] stringArray = attendance.split("\\s");
-        String date = stringArray[0];
-        Label presentLabel = new Label(date);
+    public static Label createAttendanceTag(AttendanceEntry attendanceEntry) {
+        Optional<Boolean> isPresent = attendanceEntry.getIsPresent();
+        String dateString = attendanceEntry.getAttendanceDate()
+            .format(ATTENDANCE_DATE_GUI_FORMATTER);
+
+        Label attendanceTag = new Label(dateString);
+
+        if (isPresent.isEmpty()) {
+            attendanceTag.setStyle("-fx-background-color: #c4c4c4");
+            return attendanceTag;
+        }
+
+        if (isPresent.get()) {
```

Comment:
No problem here! I just realise you use a lot of if-else statements to handle cases. Maybe here for even more readability can consider extracting the background color out to set using conditionals. For example, `String backgroundColor = [use conditionals or if-else to set color]` and then `attendanceTag.setStyle("..." + backgroundColor);`

Reply:
Will do so!